src/psyclone/tests/dependency_test.py

```diff
@@ -0,0 +1,358 @@
+                                 end program test_prog''')
+    ast = parser(reader)
+    psy = PSyFactory(API).create(ast)
+    schedule = psy.invokes.get("test_prog").schedule
+
+    code_block = schedule.children[0]
+    call_stmt = code_block.statements[0]
+    var_accesses = VariablesAccessInfo()
+    call_stmt.reference_accesses(var_accesses)
+    assert str(var_accesses) == "a: UNKNOWN, b: UNKNOWN"
+
+
+def test_do_loop(parser):
+    ''' Check the handling of do loops.
+    TODO #400: this only tests a nemo do loop. At this stage the loop
+    boundaries in nemo are only strings (not instances of Reference or so),
+    so in case of a loop like: "do jj=1, n", `n` is not be listed as
+    a READ access.
+    '''
+    reader = FortranStringReader('''program test_prog
+                                 do jj=1, 10
+                                    do ji=1, 10
+                                       s(ji, jj)=t(ji, jj)+1
+                                    enddo
+                                 enddo
+                                 end program test_prog''')
+    ast = parser(reader)
+    psy = PSyFactory(API).create(ast)
+    schedule = psy.invokes.get("test_prog").schedule
+
+    do_loop = schedule.children[0]
+    assert isinstance(do_loop, nemo.NemoLoop)
+    var_accesses = VariablesAccessInfo()
+    do_loop.reference_accesses(var_accesses)
+    assert str(var_accesses) == "ji: READWRITE, jj: READWRITE, s: WRITE, "\
+                                "t: READ"
+
+
+@pytest.mark.xfail(reason="NEMO API converts all loop limits to strings")
+def test_do_loop_not_working_yet(parser):
+    ''' Check the handling of do loops.
+    TODO #400: At this stage the loop boundaries in nemo are only strings
+    (not instances of Reference or so), so lower or upper loop boundaries
+    are not reported as 'READ'.
+    '''
+    reader = FortranStringReader('''program test_prog
+                                 do jj=1, n
+                                    do ji=1, 10
+                                       s(ji, jj)=t(ji, jj)+1
+                                    enddo
+                                 enddo
+                                 end program test_prog''')
+    ast = parser(reader)
+    psy = PSyFactory(API).create(ast)
+    schedule = psy.invokes.get("test_prog").schedule
+
+    do_loop = schedule.children[0]
+    assert isinstance(do_loop, nemo.NemoLoop)
+    var_accesses = VariablesAccessInfo()
+    do_loop.reference_accesses(var_accesses)
+    # TODO #400: n is not reported at the moment
+    assert str(var_accesses) == "ji: READWRITE, jj: READWRITE, n: READ, "\
+                                "s: WRITE, t: READ"
+
+
+@pytest.mark.xfail(reason="Gocean loops boundaries are also strings #400/#444")
+def test_goloop():
+    ''' Check the handling of non-NEMO do loops.
+    TODO #400/#444: Does not work atm, GOLoops also have start/stop as
+    strings, which are even not defined. Only after genCode will they be
```

Comment:
"after gen_code() is called will they..."

Reply:
Done.